scripts/tests/linux/log_line_processing.py

```diff
@@ -48,41 +46,48 @@ def __init__(self, command: List[str], output_path: str):
         self.output_path = output_path
         self.output_lines = queue.Queue()
         self.process = None
-        self.io_thread = None
+        self.stdout_thread = None
+        self.stderr_thread = None
+        self.lock = threading.Lock()
         self.done = False
 
-    def _io_thread(self):
-        """Reads process lines and writes them to an output file.
+    def _write_to_file(self, line: str, is_error_line=False):
+        """Writes a line to an output file in a thread-safe manner.
+
+        Opens file in append-text mode ('at') with line buffering 
+        for immediate output to ensure all writes are appended.
+        """
+        with self.lock:
+            with open(self.output_path, "at", buffering=1) as f:
```

Comment:
you are re-opening this file at every line, this seems not ideal. Can't we keep the file open and just lock around f.write() ?

Reply:
Updated in the latest commit.